### PR TITLE
Fixes cigars being lightable by everything

### LINKED ISSUE #24219: Cigars can be lit by anything
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1623

### Issue Description

Cigars can be lit by cheap lighters, despite saying they can't be.

### What did you expect to happen?

The cigar to reject the cheap lighter.

### What happened instead?

It got lit anyway.

### Why is this bad/What are the consequences?

Bugs are bad.

### Steps to reproduce the issue.

Light a cigar with a cheap lighter, watch as it tells you it rejects it while simultaneously not.

### When did the problem start happening?

Recently.  I do not remember this being a problem a few months ago.

### Extra information

N/A

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Makes cigars not being lit by everything
Resolves #24219

## Why It's Good For The Game
Cigars can only be lit by civilized means.

## Testing
Couldn't light a cigar with a cheap ligher, could with a zippo

## Changelog
:cl:
fix: Fixed cigars being lightable by everything that emitted heat
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/cigs.dm [hunk 1]
```diff
@@ -327,15 +327,25 @@ LIGHTERS ARE IN LIGHTERS.DM
...
 
+/obj/item/clothing/mask/cigarette/cigar/proc/can_light_cigar(obj/item/lighting_item)
+	if(istype(lighting_item, /obj/item/match) || istype(lighting_item, /obj/item/lighter/zippo))
+		return TRUE
+	return FALSE
+
+/obj/item/clothing/mask/cigarette/cigar/can_light(obj/item/cigar, obj/item/lighting_item)
+	if(lighting_item.get_heat() && can_light_cigar(lighting_item))
+		light()
+		return COMPONENT_CANCEL_ATTACK_CHAIN
```
Comment (Contrabang) [inline code/game/objects/items/weapons/cigs.dm:338]:
```suggestion
	if(can_light_cigar(lighting_item))
		return ..()
```

File: code/game/objects/items/weapons/cigs.dm [hunk 2]
```diff
@@ -327,15 +327,25 @@ LIGHTERS ARE IN LIGHTERS.DM
 	chem_volume = 120
 	list_reagents = list("nicotine" = 120)
 
+/obj/item/clothing/mask/cigarette/cigar/proc/can_light_cigar(obj/item/lighting_item)
+	if(istype(lighting_item, /obj/item/match) || istype(lighting_item, /obj/item/lighter/zippo))
+		return TRUE
+	return FALSE
```
Comment (Contrabang) [inline code/game/objects/items/weapons/cigs.dm:333]:
```suggestion
	return (istype(lighting_item, /obj/item/match) || istype(lighting_item, /obj/item/lighter/zippo))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Contrabang — APPROVED** [review]

File: code/game/objects/items/weapons/cigs.dm [hunk 3]
```diff
@@ -327,15 +327,22 @@ LIGHTERS ARE IN LIGHTERS.DM
 	chem_volume = 120
 	list_reagents = list("nicotine" = 120)
 
+/obj/item/clothing/mask/cigarette/cigar/proc/can_light_cigar(obj/item/lighting_item)
+	return (istype(lighting_item, /obj/item/match) || istype(lighting_item, /obj/item/lighter/zippo))
+
+/obj/item/clothing/mask/cigarette/cigar/can_light(obj/item/cigar, obj/item/lighting_item)
+	if(can_light_cigar(lighting_item))
+		return ..()
+
```
Comment (Contrabang) [inline code/game/objects/items/weapons/cigs.dm:336]: Actually, do we want to do this too?
```suggestion
	if(can_light_cigar(lighting_item))
		return ..()
	if(lighting_item.get_heat())
		to_chat(user, "<span class='notice'>[src] straight out REFUSES to be lit by such uncivilized means.</span>")
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Contrabang — APPROVED** [review]
lgtm

**Review by S34NW — APPROVED** [review]

**Pull request merged**